Release tests fail with "external test module [test-deprecated-query] found in non-snapshot build"

**Build scan**:

* https://gradle-enterprise.elastic.co/s/azaxc4dn7qcb4
* https://gradle-enterprise.elastic.co/s/cmn5yc3cjkgq4

**Repro line**:

There is no reproduction line emitted by the test infrastructure but the test that is failing in CI can be executed with:

```
./gradlew :x-pack:plugin:async-search:qa:rest:yamlRestTest
```

**Reproduces locally?**: no

**Applicable branches**: `7.x`, `master`

**Failure history**:

This started failing on November 10 and has [failed 26 times](https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:'2020-11-10T02:48:00.028Z',mode:absolute,to:'2020-11-11T08:32:21.255Z'))&_a=(columns:!(branch),index:f0e86d70-7efd-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:'log:%22external%20test%20module%22'),sort:!(time,desc))) since then.

**Failure excerpt**:

```
01:40:17 » ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [yamlRestTest-0] uncaught exception in thread [main]
01:40:17 »  org.elasticsearch.bootstrap.StartupException: java.lang.IllegalStateException: external test module [test-deprecated-query] found in non-snapshot build
01:40:17 »  	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:174) ~[elasticsearch-8.0.0.jar:8.0.0]
01:40:17 »  	at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:161) ~[elasticsearch-8.0.0.jar:8.0.0]
01:40:17 »  	at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:86) ~[elasticsearch-8.0.0.jar:8.0.0]
01:40:17 »  	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:127) ~[elasticsearch-cli-8.0.0.jar:8.0.0]
01:40:17 »  	at org.elasticsearch.cli.Command.main(Command.java:90) ~[elasticsearch-cli-8.0.0.jar:8.0.0]
01:40:17 »  	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:126) ~[elasticsearch-8.0.0.jar:8.0.0]
01:40:17 »  	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:92) ~[elasticsearch-8.0.0.jar:8.0.0]
01:40:17 »  Caused by: java.lang.IllegalStateException: external test module [test-deprecated-query] found in non-snapshot build
01:40:17 »  	at org.elasticsearch.plugins.PluginsService.findBundles(PluginsService.java:361) ~[elasticsearch-8.0.0.jar:8.0.0]
01:40:17 »  	at org.elasticsearch.plugins.PluginsService.getModuleBundles(PluginsService.java:341) ~[elasticsearch-8.0.0.jar:8.0.0]
01:40:17 »  	at org.elasticsearch.plugins.PluginsService.<init>(PluginsService.java:136) ~[elasticsearch-8.0.0.jar:8.0.0]
01:40:17 »  	at org.elasticsearch.node.Node.<init>(Node.java:333) ~[elasticsearch-8.0.0.jar:8.0.0]
01:40:17 »  	at org.elasticsearch.node.Node.<init>(Node.java:276) ~[elasticsearch-8.0.0.jar:8.0.0]
01:40:17 »  	at org.elasticsearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:225) ~[elasticsearch-8.0.0.jar:8.0.0]
01:40:17 »  	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:225) ~[elasticsearch-8.0.0.jar:8.0.0]
01:40:17 »  	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:387) ~[elasticsearch-8.0.0.jar:8.0.0]
01:40:17 »  	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:170) ~[elasticsearch-8.0.0.jar:8.0.0]
01:40:17 »  	... 6 more
```

I believe this has been introduced by #64832 as that commit has added `test-deprecated-query` which is showing up in the error message. @rjernst could you please take a look at this?

--- a/x-pack/plugin/async-search/qa/rest/build.gradle
+++ b/x-pack/plugin/async-search/qa/rest/build.gradle
@@ -2,7 +2,7 @@ apply plugin: 'elasticsearch.esplugin'
 apply plugin: 'elasticsearch.yaml-rest-test'
 
 esplugin {
-  name 'test-deprecated-query'
+  name 'x-pack-test-deprecated-query'
   description 'Deprecated query plugin'
   classname 'org.elasticsearch.query.DeprecatedQueryPlugin'
 }

